client/state/data-layer/wpcom/read/site/posts/follow/index.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * @format
+ */
+
+/**
+ * External Dependencies
+ */
+import { merge } from 'lodash';
+import { translate } from 'i18n-calypso';
+
+/**
+ * Internal Dependencies
+ */
+import { READER_CONVERSATION_FOLLOW } from 'state/action-types';
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import { errorNotice, successNotice } from 'state/notices/actions';
+import { updateConversationFollowStatus } from 'state/reader/conversations/actions';
+import { bypassDataLayer } from 'state/data-layer/utils';
+
+export function requestConversationFollow( { dispatch }, action ) {
+	const actionWithRevert = merge( {}, action, {
+		meta: {
+			// @todo once we've added convo follow to Redux state, use selector to get previous state
+			// hardcoded for the moment to support tests
+			previousState: 'M',
+		},
+	} );
+	dispatch(
+		http( {
+			method: 'POST',
+			apiNamespace: 'wpcom/v2',
+			path: `/read/sites/${ action.payload.blogId }/posts/${ action.payload.postId }/follow`,
+			body: {}, // have to have an empty body to make wpcom-http happy
+			onSuccess: actionWithRevert,
+			onFailure: actionWithRevert,
+		} )
```

Comment:
note you can pass this as a second prop to `http` since it's the same for both responders…

```js
http( { … }, actionWithRevert );
```

Reply:
Nice! I didn't know about that.